src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -52,7 +52,9 @@
     public static void writeReads(
             final JavaSparkContext ctx, final String outputFile, final String referenceFile, final JavaRDD<GATKRead> reads,
             final SAMFileHeader header, ReadsWriteFormat format) throws IOException {
-        writeReads(ctx, outputFile, referenceFile, reads, header, format, 0, null);
+        // NOTE, we must include 'format==ReadsWriteFormat.SINGLE' to preserve the old default behavior for writing spark output
+        // which would not sort the bam if outputting to ReadsWriteFormat.SINGLE. Please use the overload for different sroting behavior.
+        writeReads(ctx, outputFile, referenceFile, reads, header, format, 0, null, format==ReadsWriteFormat.SINGLE);
```

Comment:
I'd like to remove this particular wierdness.  How hard is it to just eliminate this behavior?

Reply:
I think the only remaining use is in BwaSpark and that can be changed to just always sort I think.

Reply:
Or to take a parameter to sort I guess.

Reply:
Either way...